singer/src/main/java/com/pinterest/singer/writer/headersinjectors/LoggingAuditHeadersInjector.java

```diff
@@ -0,0 +1,40 @@
+package com.pinterest.singer.writer.headersinjectors;
+
+import com.pinterest.singer.common.SingerMetrics;
+import com.pinterest.singer.thrift.LogMessage;
+import com.pinterest.singer.writer.HeadersInjector;
+
+import com.twitter.ostrich.stats.Stats;
+import org.apache.kafka.common.header.Headers;
+import org.apache.thrift.TException;
+import org.apache.thrift.TSerializer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * LoggingAuditHeadersInjector can extract LoggingAuditHeaders from LogMessage and inject this
+ * this object as the headers of ProducerRecord's Headers.
+ */
+public class LoggingAuditHeadersInjector implements HeadersInjector {
+
+  private static final Logger LOG = LoggerFactory.getLogger(LoggingAuditHeadersInjector.class);
+  private static final String HEADER_KEY = "loggingAuditHeaders";
```

Comment:
shall we shorten the string as `auditHeader` to save some bytes?

Reply:
I feel loggingAuditHeaders is better.  the reason is that: 
in Singer, we still have some legacy audit feature (will be removed after we have 0.8.0 released and running stable).  Using loggingAudit as prefix for many variables and constants (including the one we are discussing) is to differentiate from the previous audit code.  Most of the extra storage for headers is the actual content of loggingAuditHeaders (byte[]) and the HEADER_KEY name is a much smaller portion.